### ISSUE TITLE
Filter potential updates, rather than current dependencies

### ISSUE THREAD

**Issue by barnardb** [rtimush/sbt-updates#44]
I was hoping to be able to use `dependencyUpdatesExclusions` to exclude scala-library updates when they are for a different Scala binary version than the current one, and I have a cross Scala version relevant to the update in question. Concretely, I had a build with:

~~~ {.sbt}
crossScalaVersions in Global := Seq("2.10.4", "2.11.5")
~~~

and ran `+dependencyUpdates`, which for 2.10.4 told me I could update

    org.scala-lang:scala-library : 2.10.4 -> 2.10.5 -> 2.11.6

and for 2.11.5 that I could do

    org.scala-lang:scala-library : 2.11.5 -> 2.11.6

Awesome. After updating to `Seq("2.10.5", "2.11.6")`, `dependencyUpdates` for 2.10.5 now shows `2.10.5 -> 2.11.6`, which is annoying.

I thought I could use the new `dependencyUpdatesExclusions` setting to ignore this update, like this:

~~~ {.sbt}
dependencyUpdatesExclusions :=
  moduleFilter(
    organization = ScalaArtifacts.Organization,
    name = ScalaArtifacts.LibraryID,
    revision = { version: String =>
      val binaryVersion = CrossVersion.binaryScalaVersion(version)
      scalaBinaryVersion.value != binaryVersion &&
        crossScalaVersions.value.map(CrossVersion.binaryScalaVersion).contains(binaryVersion)
    }
  )
~~~

But it turns out that the revision this filter sees is not for the potential *update* as I originally thought (in which case it would see the available "2.11.6" revision), but rather the version of the *existing* dependency (so it sees "2.10.5"). There is thus no way for me to write a filter that would show me the two potential updates (2.10.4 -> 2.10.5 and 2.11.5 -> 2.11.6) before I updated my project, and not try to suggest 2.10.5 -> 2.11.6 after the update.

Something like `dependencyUpdatesExclusions` that sees the potential updates rather than the current dependencies would be more useful in this case. It would also allow rules like "ignore updates to this library that have major version 0 (such as 0.3.1, 0.4.0, …), but report when version 1 becomes available".

**Comment by rtimush** [comment]
Thank you for pointing this out. This really makes sense, I will fix it.

**Comment by rtimush** [comment]
Fixed in 818e756b4b4620a9f044b5859afc45755db0989c and published as `0.1.9-SNAPSHOT`

**Comment by rtimush** [comment]
I tried the following build.sbt:
```sbt
crossScalaVersions in Global := Seq("2.10.4", "2.11.5")

dependencyUpdatesExclusions :=
  moduleFilter(
    organization = ScalaArtifacts.Organization,
    name = ScalaArtifacts.LibraryID,
    revision = { version: String =>
      val binaryVersion = CrossVersion.binaryScalaVersion(version)
      scalaBinaryVersion.value != binaryVersion &&
        crossScalaVersions.value.map(CrossVersion.binaryScalaVersion).contains(binaryVersion)
    }
  )
```
And the result is 
```
$ sbt dependencyUpdates
...
[info] Found 1 dependency update for cross
[info]   org.scala-lang:scala-library : 2.10.4 -> 2.10.5
...
```
Which looks ok. @barnardb, can you please check that your sbt-updates snapshot is up-to-date?

**Comment by barnardb** [comment]
Yes, looks good to me now as well.

Cheers. :)

**Comment by barnardb** [comment]
The current snapshot has been working quite well for me for several months now, but it would be more convenient to have a non-snapshot version. Is there any possibility of a 0.1.9 release?

**Comment by rtimush** [comment]
Yes, I'll release 0.1.9 by the end of this week.

**Comment by rtimush** [comment]
Published 0.1.9, sorry for a delay
